nsxt/resource_nsxt_policy_context_profile_custom_attribute.go

```diff
@@ -0,0 +1,209 @@
+}
+
+func resourceNsxtPolicyContextProfileCustomAttribute() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNsxtPolicyContextProfileCustomAttributeCreate,
+		Read:   resourceNsxtPolicyContextProfileCustomAttributeRead,
+		Delete: resourceNsxtPolicyContextProfileCustomAttributeDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"key": {
+				Type:         schema.TypeString,
+				Description:  "Key for attribute",
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringInSlice(customAttributeKeys, false),
+			},
+			"attribute": {
+				Type:        schema.TypeString,
+				Description: "Custom Attribute",
+				Required:    true,
+				ForceNew:    true,
+			},
+		},
+	}
+}
+
+func resourceNsxtPolicyContextProfileCustomAttributeExists(id string, connector *client.RestConnector, isGlobalmodel bool) (bool, error) {
+	var err error
+	var attrList model.PolicyContextProfileListResult
+
+	key, attribute := splitCustomAttributeID(id)
+	source := model.PolicyCustomAttributes_ATTRIBUTE_SOURCE_CUSTOM
+	if isGlobalmodel {
+		var gmAttrList gm_model.PolicyContextProfileListResult
+		client := gm_infra.NewDefaultClient(connector)
+		gmAttrList, err = client.List(&key, &source, nil, nil, nil, nil, nil, nil)
+
+		al, err1 := convertModelBindingType(gmAttrList, gm_model.PolicyContextProfileListResultBindingType(), model.PolicyContextProfileListResultBindingType())
+		if err1 != nil {
+			return false, err1
+		}
+
+		attrList = al.(model.PolicyContextProfileListResult)
+	} else {
+		client := infra.NewDefaultClient(connector)
+		attrList, err = client.List(&key, &source, nil, nil, nil, nil, nil, nil)
+		if err != nil {
+			return false, err
+		}
+	}
+
+	if *attrList.ResultCount == 0 {
+		return false, nil
+	}
+	if isNotFoundError(err) {
+		return false, nil
+	}
+
+	if err == nil {
+		for _, c := range attrList.Results {
+			for _, a := range c.Attributes {
+				if *a.Key == key {
+					for _, v := range a.Value {
+						if v == attribute {
+							return true, nil
+						}
+					}
+				}
+			}
+		}
+	}
+
+	return false, err
+}
+
+func resourceNsxtPolicyContextProfileCustomAttributeRead(d *schema.ResourceData, m interface{}) error {
+	id := d.Id()
+	key, attribute := splitCustomAttributeID(id)
+
+	log.Printf("[INFO] Reading ContextProfileCustomAttribute with ID %s", d.Id())
+
+	connector := getPolicyConnector(m)
+	exists, err := resourceNsxtPolicyContextProfileCustomAttributeExists(id, connector, isPolicyGlobalManager(m))
+	if err != nil {
+		return err
+	}
+	if !exists {
+		return errors.NotFound{}
+	}
+	d.Set("key", key)
+	d.Set("attribute", attribute)
+	return nil
+}
+
+func resourceNsxtPolicyContextProfileCustomAttributeCreate(d *schema.ResourceData, m interface{}) error {
+	var err error
+	key := d.Get("key").(string)
+	attribute := d.Get("attribute").(string)
+	if d.Id() == "" {
+		d.SetId(makeCustomAttributeID(key, attribute))
```

Comment:
I believe we need to SetId at the end of Create, once we're the resource was created succesfully

Reply:
OK